lib/CXGN/Stock/AddDerivedAccession.pm

```diff
@@ -0,0 +1,270 @@
+=head1 NAME
+
+CXGN::Stock::AddDerivedAccession - a module for adding new accession derived from another stock type
+
+=cut
+
+
+package CXGN::Stock::AddDerivedAccession;
+
+use Moose;
+use MooseX::FollowPBP;
+use Moose::Util::TypeConstraints;
+use Try::Tiny;
+use CXGN::Location::LocationLookup;
+use SGN::Model::Cvterm;
+use Data::Dumper;
+use CXGN::Stock;
+
+has 'chado_schema' => (
+    isa => 'DBIx::Class::Schema',
+    is => 'rw',
+    required => 1,
+);
+
+has 'dbh' => (
+    is  => 'rw',
+    required => 1,
+);
+
+has 'phenome_schema' => (
+    is => 'rw',
+    isa => 'DBIx::Class::Schema',
+    predicate => 'has_phenome_schema',
+    required => 1,
+);
+
+has 'derived_accession_name' => (
+    isa =>'Str',
+    is => 'rw',
+    required => 1,
+);
+
+has 'derived_from_stock_id' => (
+    isa => 'Int',
+    is => 'rw',
+    required => 1,
+);
+
+has 'description' => (
+    isa => 'Maybe[Str]',
+    is => 'rw',
+);
+
+has 'owner_id' => (
+    isa => 'Int',
+    is => 'rw',
+    required => 1,
+);
+
+
+sub existing_accession_name {
+    my $self = shift;
+    my $schema = $self->get_chado_schema();
+    my $derived_accession_name = $self->get_derived_accession_name();
+
+    if ($schema->resultset('Stock::Stock')->find({ 'uniquename' => $derived_accession_name, 'is_obsolete' => { '!=' => 't' }})){
```

Comment:
Is having the `is_obsolete` check here a good idea?  Could this allow someone to create a derived accession with the same name as an existing obsoleted stock?  Would that cause a problem?